File: code/python/strategy/RoutePlanner.py
class RoutePlanner:
    def __init__(self):
        self.strategy = None

    def set_strategy(self, strategy):
        self.strategy = strategy
        print(f"Setting the strategy to calculate the route: {type(self.strategy).__name__}")

    def execute_strategy(self):
        print(f"Executing the strategy to calculate the route for {type(self.strategy).__name__}")
        self.strategy.calculate_route()

File: code/typescript/strategy/RoutePlanner.ts
import { IRouteStrategy } from "./RouteStrategy";

export class RoutePlanner {
    private strategy: IRouteStrategy;

    setStrategy(strategy: IRouteStrategy): void {
        console.log("Setting the strategy to calculate the route: " + strategy.constructor.name);
        this.strategy = strategy;
    }

    executeStrategy(): void {
        console.log("Executing the strategy to calculate the route for " + this.strategy.constructor.name);
        this.strategy.calculateRoute();
    }
}
